server/main.py

```
@@ -0,0 +1,280 @@
+        
+        # Also process individual DTX files for backward compatibility
+        dtx_files = []
+        for file_path in DTX_FILES_DIR.glob("*.dtx"):
+            dtx_files.append({
+                "filename": file_path.name,
+                "size": file_path.stat().st_size
+            })
+        
+        return {"songs": songs, "individual_files": dtx_files}
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Error listing DTX songs: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)